app/controllers/ctc/questions/confirm_payment_controller.rb

```diff
@@ -10,13 +10,16 @@ def edit
         benefits = Efile::BenefitsEligibility.new(tax_return: tax_return, dependents: current_intake.dependents)
         @ctc_amount = benefits.outstanding_ctc_amount
         @third_stimulus_amount = benefits.outstanding_eip3
-        @not_collecting = @ctc_amount.zero? && @third_stimulus_amount.zero?
+        @eitc_amount = benefits.claiming_and_qualified_for_eitc? ? benefits.eitc_amount : nil
+        @not_collecting = @ctc_amount.zero? && @third_stimulus_amount.zero? && @eitc_amount.zero?
```

Comment:
Blocking request: I think this crashes when `eitc_amount` is `nil`. I encourage you to compute `@total_amount` first, and then compute `@not_collecting` as `@total_amount == 0`.

Non-blocking request: What do you think about renaming the variable to be positive, like `@collecting` (rather than `@not_collecting`), or perhaps inlining the variable? I think it's only used in the template, which could just as easily check if `@total_amount > 0`. The best reason IMHO to keep the variable is to cover it in unit tests.

L13 sometimes sets `@eitc_amount` to `nil`, `@eitc_amount.zero?` will crash (see `rails c` transcript below). If `@total_amount` already uses `.compact` to handle the nil case, rely on that. I'm assuming the intent is that `@not_collecting` is `true` if `benefits.claiming_and_qualified_for_eitc?` is false and everything else is zero.

```
irb(main):001:0> nil.zero?
Traceback (most recent call last):
        1: from (irb):1
NoMethodError (undefined method `zero?' for nil:NilClass)
```

Reply:
Yes that makes sense! Going to get rid of non-collecting and just check for @total.zero? in the code